sorc/global_cycle.fd/cycle.f90

```diff
@@ -516,19 +521,38 @@ SUBROUTINE SFCDRV(LUGB, IDIM,JDIM,LSM,LENSFC,LSOIL,DELTSFC,  &
 !--------------------------------------------------------------------------------
 
  IF (DO_SFCCYCLE) THEN
+   ALLOCATE(SLMASKL(LENSFC), SLMASKW(LENSFC))
+! for running uncoupled (non-fractional grid)
+   DO I=1,LENSFC
+     IF(NINT(SLMASK(I)) == 1) THEN
+       SLMASKL(I) = 1.0_KIND_io8
+       SLMASKW(I) = 1.0_KIND_io8
+     ELSE
+       SLMASKL(I) = 0.0_KIND_io8
+       SLMASKW(I) = 0.0_KIND_io8
+     ENDIF
+     if(nint(slmask(i)) == 0) then
+       min_ice(i) = 0.15_KIND_io8
+     else
+       min_ice(i) = 0.0_KIND_io8
+     endif
```

Comment:
Is slmask=0 an ocean point? Why does min_ice depend on surface type, 15% over ocean and 0% on land?

Reply:
Originally, I tried setting min_ice to 15% everywhere. And I recall getting sea ice at all non-land points, which is clearly wrong. I am not totally sure how to set this array for the non-coupled case. Do you know how to set this array?

Reply:
In the current FV3/ccpp/data/GFS_typedefs.F90, min ice for lake and sea are set to:

    real(kind=kind_phys) :: min_lakeice     = 0.15d0          !< minimum lake ice value
    real(kind=kind_phys) :: min_seaice      = 1.0d-11         !< minimum sea  ice value

There might be a reason for min ice to have a higher value of 15% in the non-coupled case that I am not aware of. Why min_ice is an array, not a variable?